### PR TITLE
Upgrade coredns to 1.6.7

### PR DESCRIPTION
**Pull request by pnovotnak**
This allows the use of `serve_stale`. I'd like to use this to make [resolving from an upstream](https://www.consul.io/docs/platform/k8s/dns.html#coredns-configuration) consistently fast.

Builds and passes tests locally.

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -79,7 +79,7 @@
 
 [[constraint]]
   name = "github.com/coredns/coredns"
-  version = "v1.6.4"
+  version = "v1.6.7"
```
Comment (prameshj) [inline Gopkg.toml]: is this the only manual change?

**Comment by prameshj** [comment]
/ok-to-test

could you also build the container images and make sure node-local-dns pods run successfully with the new image?

File: Gopkg.toml (hunk 1 above)
Comment (pnovotnak) [reply Gopkg.toml]: Yep!

**Comment by pnovotnak** [comment]
I've tested node-local-dns. It works, and the new cache option is present.

## Test config

```
apiVersion: v1
data:
  Corefile: |
    cluster.local:53 {
        errors
        cache {
                success 9984 30
                denial 9984 5
        }
        reload
        loop
        bind 169.254.20.10
        forward . 192.168.0.10 {
                force_tcp
        }
        prometheus :9253
        health 169.254.20.10:8080
        }
    in-addr.arpa:53 {
        errors
        cache 30
        reload
        loop
        bind 169.254.20.10
        forward . 192.168.0.10 {
                force_tcp
        }
        prometheus :9253
        }
    ip6.arpa:53 {
        errors
        cache 30
        reload
        loop
        bind 169.254.20.10
        forward . 192.168.0.10 {
                force_tcp
        }
        prometheus :9253
        }
    .:53 {
        errors
        cache 30
        reload
        loop
        bind 169.254.20.10
        forward . /etc/resolv.conf {
                force_tcp
        }
        prometheus :9253
        }
    consul:53 {
        errors
        cache 30 {
                prefetch 90
                serve_stale
        }
        reload
        loop
        loadbalance round_robin
        # bind 169.254.20.10
        # IPs changed for privacy
        forward . 10.0.1.10 10.0.2.10 10.0.3.10 {
                force_tcp
        }
        prometheus :9253
        }
...
```
## Log output

```
2020/02/14 04:13:17 [INFO] Using Corefile /etc/coredns/Corefile
2020/02/14 04:13:17 [INFO] Tearing down
2020/02/14 04:13:17 [INFO] Hit error during teardown - Link not found
2020/02/14 04:13:17 [INFO] Setting up networking for node cache
2020/02/14 04:13:17 [ERROR] Failed to read node-cache coreFile /etc/coredns/Corefile.base - open /etc/coredns/Corefile.base: no such file or directory
2020/02/14 04:13:17 [ERROR] Failed to sync kube-dns config directory /etc/kube-dns, err: lstat /etc/kube-dns: no such file or directory
cluster.local.:53 on 169.254.20.10
in-addr.arpa.:53 on 169.254.20.10
ip6.arpa.:53 on 169.254.20.10
.:53 on 169.254.20.10
consul.:53
[INFO] plugin/reload: Running configuration MD5 = 9ce2021cd94d32550273d0ab031bf343
CoreDNS-1.6.7
linux/amd64, go1.11.13,
```

I can do `dig listing.service.consul @localhost` and the expected consul record is returned.

**Comment by pnovotnak** [comment]
@prameshj could you kick the build? Sounds like travis can't be kicked with `/retest`?

**Comment by prameshj** [comment]
> I've tested node-local-dns. It works, and the new cache option is present.

Perfect, thanks.

> 
> ## Test config
> ```
> apiVersion: v1
> data:
>   Corefile: |
>     cluster.local:53 {
>         errors
>         cache {
>                 success 9984 30
>                 denial 9984 5
>         }
>         reload
>         loop
>         bind 169.254.20.10
>         forward . 192.168.0.10 {
>                 force_tcp
>         }
>         prometheus :9253
>         health 169.254.20.10:8080
>         }
>     in-addr.arpa:53 {
>         errors
>         cache 30
>         reload
>         loop
>         bind 169.254.20.10
>         forward . 192.168.0.10 {
>                 force_tcp
>         }
>         prometheus :9253
>         }
>     ip6.arpa:53 {
>         errors
>         cache 30
>         reload
>         loop
>         bind 169.254.20.10
>         forward . 192.168.0.10 {
>                 force_tcp
>         }
>         prometheus :9253
>         }
>     .:53 {
>         errors
>         cache 30
>         reload
>         loop
>         bind 169.254.20.10
>         forward . /etc/resolv.conf {
>                 force_tcp
>         }
>         prometheus :9253
>         }
>     consul:53 {
>         errors
>         cache 30 {
>                 serve_stale
>         }
>         reload
>         loop
>         loadbalance round_robin

Do you need to add loadbalance plugin to node-cache?
>         # bind 169.254.20.10
>         # IPs changed for privacy
>         forward . 10.0.1.10 10.0.2.10 10.0.3.10 {
>                 force_tcp
>         }
>         prometheus :9253
>         }
> ...
> ```
> 
> ## Log output
> ```
> 2020/02/14 04:13:17 [INFO] Using Corefile /etc/coredns/Corefile
> 2020/02/14 04:13:17 [INFO] Tearing down
> 2020/02/14 04:13:17 [INFO] Hit error during teardown - Link not found
> 2020/02/14 04:13:17 [INFO] Setting up networking for node cache
> 2020/02/14 04:13:17 [ERROR] Failed to read node-cache coreFile /etc/coredns/Corefile.base - open /etc/coredns/Corefile.base: no such file or directory
> 2020/02/14 04:13:17 [ERROR] Failed to sync kube-dns config directory /etc/kube-dns, err: lstat /etc/kube-dns: no such file or directory
> cluster.local.:53 on 169.254.20.10
> in-addr.arpa.:53 on 169.254.20.10
> ip6.arpa.:53 on 169.254.20.10
> .:53 on 169.254.20.10
> consul.:53
> [INFO] plugin/reload: Running configuration MD5 = 9ce2021cd94d32550273d0ab031bf343
> CoreDNS-1.6.7
> linux/amd64, go1.11.13,
> ```
> 
> I can do `dig listing.service.consul @localhost` and the expected consul record is returned.

**Comment by pnovotnak** [comment]
Sorry, that's from a continuation of this PR in a fork. We'd love to have that in this image, but if you don't want it in the base we can build our own version.

The two plugins we want to enable in our version are `loadbalance` and `whoami`. 

- `loadbalance` is useful because of how consul returns records (large sets of A records which are randomized on retrieval). However, coredns doesn't randomize this set. So the records remain ordered until a new set is retrieved. I'm worried that this will cause hotspotting.
- `whoami` is useful because "we can add node.cluster.local as a zone that resolves to the host ip. Then we can put that value into configmaps, vs having to use downward api and hacks (since configmaps don't natively support that). Our specific use case is a node local memcached instance listening on a hostport" -@gbrayut

**Comment by pnovotnak** [comment]
Plugin additions might be better in their own PR anyhow?

**Comment by prameshj** [comment]
> Plugin additions might be better in their own PR anyhow?

yes sounds good.

**Pull request merged**